### ISSUE TITLE
Add Support For Tuples

### ISSUE THREAD

**Issue by joeblau** [imanrep/swiftabigen#1]
https://docs.soliditylang.org/en/latest/control-structures.html#assignment

When generating boilerplate for [XENCrypto](https://etherscan.io/token/0x06450dEe7FD2Fb8E39061434BAbCFC05599a6Fb8#code) it uses a tuple as one of the `types`. 

Since tuple isn't supported by the parser, generating the ABI fails.

**Comment by imanrep** [comment]
yes. i've thinking to support tupple. but i have no time for update. Thanks for remind me for this issue, i will update when i have free time

**Comment by joeblau** [comment]
Awesome. I may take a look and submit a PR after I'm done prototyping. Thanks for all of your work with this library, because I was doing this manually 🫡
